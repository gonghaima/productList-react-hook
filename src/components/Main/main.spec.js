import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { Main } from "./index";

let realUseContext;
let useContextMock;

const running = {
  status: "running"
};

const ready = {
  status: "ready"
};
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

test("Main component should be contains loader if STATUS is running", () => {
  useContextMock.mockReturnValue({ state: running });
  const element = new ShallowRenderer().render(<Main />);
  expect(element.props.children[0].props.className).toBe("split");
  expect(element).toMatchSnapshot();
});
