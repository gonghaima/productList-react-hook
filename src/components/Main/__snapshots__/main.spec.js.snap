// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should be contains loader if STATUS is running 1`] = `
<React.Suspense
  fallback={<Loader />}
>
  <hr
    className="split"
  />
  <Loader />
</React.Suspense>
`;
